src/gui/src/heatMap.cpp

```diff
@@ -340,24 +340,36 @@ void HeatMapDataSource::setSettings(const Renderer::Settings& settings)
 HeatMapDataSource::MapView HeatMapDataSource::getMapView(
     const odb::Rect& bounds)
 {
-  const double dbu_per_micron = getBlock()->getDbUnitsPerMicron();
+  const auto x_low_find = std::find_if(
+      map_x_grid_.begin(), map_x_grid_.end(), [&bounds](const int x) {
+        return x >= bounds.xMin();
+      });
+  const auto x_high_find = std::find_if(
+      map_x_grid_.begin(), map_x_grid_.end(), [&bounds](const int x) {
+        return x > bounds.xMax();
+      });
```

Comment:
Does this linear search hurt runtime?  I'm wondering if binary search is needed.

For high you can start from where you found low rather than the start.

Reply:
I didn't notice much, but that is not to say on a larger design it would be apparent.
I'm open to suggestions if you have any.

Reply:
std::lower_bound

Reply:
I could try it out and see